vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java

```diff
@@ -245,27 +239,33 @@ protected void launch() {
       return;
     }
 
-    final String oldSide = getMySide();
-
     String newSide;
     newSide = promptForSide();
-    if (newSide == null) {
+    if ((newSide == null) || newSide.equals(mySide)) {
       return;
     }
 
-    remove(GameModule.getUserId());
+    final GameModule gm = GameModule.getGameModule();
 
     final PlayerInfo me = new PlayerInfo(
       GameModule.getUserId(),
       GlobalOptions.getInstance().getPlayerId(),
       newSide
     );
-    final Add a = new Add(this, me.playerId, me.playerName, me.side);
+
+    Command c = new Chatter.DisplayText(gm.getChatter(), Resources.getString(GlobalOptions.getInstance().chatterHTMLSupport() ? "PlayerRoster.changed_sides_2" : "PlayerRoster.changed_sides", me.playerName, mySide, newSide));
+    c.execute();
+
+    remove(GameModule.getUserId());
+
+    Add a = new Add(this, me.playerId, me.playerName, me.side);
     a.execute();
-    GameModule.getGameModule().getServer().sendToOthers(a);
+
+    c = c.append(a);
+    gm.getServer().sendToOthers(c);
 
     newSide = getMySide();
-    fireSideChange(oldSide, newSide);
+    fireSideChange(mySide, newSide);
```

Comment:
How does this work? Why can we get rid of `oldSide`?

Reply:
Since I discovered the code for that method looked like this: :)
```java
protected void launch() {
    final String mySide = getMySide();
    if (mySide == null && allSidesAllocated()) {
      return;
    }

    final String oldSide = getMySide();
```

Reply:
Derp.